include/dos_inc.h

```diff
@@ -560,29 +574,45 @@ class DOS_DTA:public MemStruct{
 	#endif
 };
 
-class DOS_FCB: public MemStruct {
+/* File Control Block */
+
+class DOS_FCB : public MemStruct {
 public:
-	DOS_FCB(Bit16u seg,Bit16u off,bool allow_extended=true);
+	DOS_FCB(uint16_t seg, uint16_t off, bool allow_extended = true);
+
 	void Create(bool _extended);
-	void SetName(Bit8u _drive,char * _fname,char * _ext);
-	void SetSizeDateTime(Bit32u _size,Bit16u _date,Bit16u _time);
-	void GetSizeDateTime(Bit32u & _size,Bit16u & _date,Bit16u & _time);
+
+	bool Extended() const { return extended; }
+
+	void SetName(uint8_t drive, const char *fname, const char *ext);
 	void GetName(char * fillname);
-	void FileOpen(Bit8u _fhandle);
-	void FileClose(Bit8u & _fhandle);
-	void GetRecord(Bit16u & _cur_block,Bit8u & _cur_rec);
-	void SetRecord(Bit16u _cur_block,Bit8u _cur_rec);
-	void GetSeqData(Bit8u & _fhandle,Bit16u & _rec_size);
-	void SetSeqData(Bit8u _fhandle,Bit16u _rec_size);
-	void GetRandom(Bit32u & _random);
-	void SetRandom(Bit32u  _random);
-	Bit8u GetDrive(void);
-	bool Extended(void);
-	void GetAttr(Bit8u & attr);
-	void SetAttr(Bit8u attr);
+
+	void SetSizeDateTime(uint32_t size, uint16_t date, uint16_t time);
+	void GetSizeDateTime(uint32_t &size, uint16_t &date, uint16_t &time) const;
+
+	void FileOpen(uint8_t fhandle);
+	void FileClose(uint8_t &fhandle);
```

Comment:
Pedantically, almost all functions in these classes wrapping DOS structures are const (because they do not modify memory pointed by `this` - they modify memory pointed bl DOSBox "pointer" `pt`.

As a matter of keeping the interface sane (in case we'll want to refactor these classes in the future), I decided to add `const` only to getters / methods utilizing `SGET_*` macros. Both `FileOpen` and `FileClose` use `SSET_*` macros.

In other words - in these classes I decided to make `const` synonymous to "this function has no side effects", otherwise code like this would be valid, and it might cause problems down the road:
```
const DOS_SomeStruct ds;
ds.SetField(value);
```